pkg/client/checkpointer_test.go

```diff
@@ -0,0 +1,142 @@
+/*
+Copyright 2022 IBM All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package client
+
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+var path = "checkpointer_test_file.json"
+
+type checkpoint interface {
+	CheckpointBlock(uint64) error
+	CheckpointTransaction(uint64, string) error
+	CheckpointChaincodeEvent(*ChaincodeEvent) error
+}
+
+type Checkpoint interface {
+	checkpoint
+	Checkpointer
+}
+
+type testCase struct {
+	description     string
+	after           func()
+	newCheckpointer func() Checkpoint
```

Comment:
Since this can error for a file checkpointer, perhaps `newCheckpointer` should return an error, which would always be `nil` for the in-memory checkpointer?
```suggestion
	newCheckpointer func() (Checkpoint, error)
```

Or if you never want to test error conditions, you could pass the Go test in to allow this to fail the test on error rather than ignore it:
```suggestion
	newCheckpointer func(t *testing.T) Checkpoint
```